### PR TITLE
update to secp256k1 0.10.0

### PR DESCRIPTION
**Pull request by apoelstra**

### CODE REVIEW

**Comment by apoelstra** [comment]
cc @TheBlueMatt

**Comment by apoelstra** [comment]
Rebased.

File: src/util/contracthash.rs [hunk 1]
```diff
@@ -194,9 +194,9 @@ pub fn compute_tweak(secp: &Secp256k1, pk: &PublicKey, contract: &[u8]) -> Resul
 }
 
 /// Tweak a secret key using some arbitrary data (calls `compute_tweak` internally)
-pub fn tweak_secret_key(secp: &Secp256k1, key: &SecretKey, contract: &[u8]) -> Result<SecretKey, Error> {
+pub fn tweak_secret_key<C: secp256k1::Signing + secp256k1::Verification>(secp: &Secp256k1<C>, key: &SecretKey, contract: &[u8]) -> Result<SecretKey, Error> {
```
Comment (TheBlueMatt) [inline src/util/contracthash.rs]: This doesn't need Verification.

Comment (apoelstra) [reply src/util/contracthash.rs]: Good catch. Fixed.

**Pull request merged**